src/index.js

```diff
@@ -0,0 +1,148 @@
+
+  };
+
+  setLocationObject(currentLoc, myCoordsObj);
+  updateDataAndDisplay(currentLoc);
+};
+
+const loadWeather = (event) => {
+  const savedLocation = getHomeLocation();
+  if (!savedLocation && !event) return getGeoWeather();
+  if (!savedLocation && event.type === 'click') {
+    displayError(
+      'No Home Location Saved.',
+      'Sorry. Please save your home location first.',
+    );
+  } else if (savedLocation && !event) {
+    displayHomeLocationWeather(savedLocation);
+  } else {
+    const homeIcon = document.querySelector('.fa-home');
+    displayHomeLocationWeather(savedLocation);
+  }
+};
+const displayHomeLocationWeather = (home) => {
+  if (typeof home === 'string') {
+    const locationJson = JSON.parse(home);
+    const myCoordsObj = {
+      lat: locationJson.lat,
+      lon: locationJson.lon,
+      name: locationJson.name,
+      unit: locationJson.unit,
+    };
+    setLocationObject(currentLoc, myCoordsObj);
+    updateDataAndDisplay(currentLoc);
+  }
+};
+
+const saveLocation = () => {
+  if (currentLoc.getLat() && currentLoc.getLon()) {
+    const saveIcon = document.querySelector('.fa-save');
+
+    const location = {
+      name: currentLoc.getName(),
+      lat: currentLoc.getLat(),
+      lon: currentLoc.getLon(),
+      unit: currentLoc.getUnit(),
+    };
+    localStorage.setItem('defaultWeatherLocation', JSON.stringify(location));
+    updateScreenReaderConfirmation(
+      `Saved ${currentLoc.getName()} as home location.`,
+    );
+  }
+};
+
+const setUnitPref = () => {
+  const unitIcon = document.querySelector('.fa-chart-bar');
+  // addSpinner(unitIcon);
+  currentLoc.toggleUnit();
+  updateDataAndDisplay(currentLoc);
+};
+
+const refreshWeather = () => {
+  const refreshIcon = document.querySelector('.fa-sync-alt');
+  // addSpinner(refreshIcon);
+  updateDataAndDisplay(currentLoc);
+};
+
+const submitNewLocation = async (event) => {
+  event.preventDefault();
+  const text = document.getElementById('searchBar_text').value;
+  const entryText = cleanText(text);
+  if (!entryText.length) return;
+  const locationIcon = document.querySelector('.fa-search');
+  // addSpinner(locationIcon);
+  const coordsData = await getCoordsFromApi(entryText, currentLoc.getUnit());
+  if (coordsData) {
+    if (coordsData.cod === 200) {
+      const myCoordsObj = {
+        lat: coordsData.coord.lat,
+        lon: coordsData.coord.lon,
+        name: coordsData.sys.country
+          ? `${coordsData.name}, ${coordsData.sys.country}`
+          : coordsData.name,
+      };
+      setLocationObject(currentLoc, myCoordsObj);
+      updateDataAndDisplay(currentLoc);
+    } else {
+      displayApiError(coordsData);
+    }
+  } else {
+    displayError('Connection Error', 'Connection Error');
+  }
+};
+
+const updateDataAndDisplay = async (locationObj) => {
+  const weatherJson = await getWeatherFromCoords(locationObj);
+  if (weatherJson) updateDisplay(weatherJson, locationObj);
+  console.log(weatherJson);
```

Comment:
- [ ] It will be great to delete all `console.log` statements in your application. This is because it kills performance and it can make confidential data be visible in the browser tools. ⬇️ 
![console_log_requirement](https://user-images.githubusercontent.com/32361963/126021398-1cc3081d-70ca-4acb-8645-a31be79a2f75.png)